dataService.js

```diff
@@ -0,0 +1,94 @@
+// Maps
+let productMap;
+let categoryMap;
+let combinedProductMap;
+
+/*
+ * Method to create and return map of products if its not already created
+ */
+function getProducts() {
+
+  if(productMap) {
+    return productMap;
+  }
+
+  const productData = require('./src/data/products.json');
+
+  // Method variable to be stored in productMap
+  function createProductMap() {
```

Comment:
Do we actually need this method inside another method? Can we just put it inside the parent method?

Reply:
Actually I did it that way but you suggested me to use method variable, but I think we should do it this way because assigning a function makes it possible to adapt changes in original data like if new data is added/deleted, the function can change the map accordingly.

So I think we should do it this way :)

Reply:
Hii, I changed it to the way you suggested because we don't really need that in this level